libs/designer/src/lib/ui/panel/nodeDetailsPanel/tabs/parametersTab/index.tsx

```diff
@@ -413,11 +415,26 @@ const ParameterSection = ({
   );
 };
 
-const getEditorAndOptions = (
+export const getEditorAndOptions = (
+  operationInfo: OperationInfo,
   parameter: ParameterInfo,
   upstreamNodeIds: string[],
   variables: Record<string, VariableDeclaration[]>
 ): { editor?: string; editorOptions?: any } => {
+  const customEditor = EditorService()?.getEditor({
```

Comment:
let's move this filtering logic outside into our into getEditor, and let's make it so that getEditor returns { editor?: string; editorOptions?: any }, so we won't need toEditorAndOptions

Reply:
Instead of destructuring parameter and operationInfo, we can just pass them the whole object. What I'm getting at is that we shouldn't be creating new interfaces that abstract the Parameter Information, and if there is more information that you want to be returned from the customer editor you can extend it there (for now I know there will be circular dependencies, but for now can we just create a copy of the interfaces -- a future PR I'll extract these models to the utils that can be imported). Also I think most of this logic should be done in libs\designer\src\lib\core\utils\parameters\helper.ts at: 
```
getParameterEditorProps
```
We shouldn't be calling our services in our UI component